File: gridfly-quickstart-dot-net/AsyncApiClient.cs
using System;
using System.IO;
using System.Threading;
using System.Threading.Tasks;
using Gridfly.Client;

namespace Gridfly.QuickStart
{
    public class AsyncApiClient : BaseApiClient
    {
        public AsyncApiClient(string clientId, string clientSecret) : base(clientId, clientSecret)
        {
        }

        public async Task Export(Stream htmlStream)
        {
            var client = await Authenticate();
            JobCreationResponse asyncResponse;
            try
            {
                asyncResponse = await client.CreateAsyncJobAsync(false, true, htmlStream);
            }
            catch (ApiException ex) when (ex.StatusCode == 201)
            {
                asyncResponse = Newtonsoft.Json.JsonConvert.DeserializeObject<JobCreationResponse>(ex.Response!)!;
            }
            catch (ApiException ex)
            {
                Console.WriteLine($"API returned error with status code {ex.StatusCode}:");
                try {
                    var parsed = Newtonsoft.Json.JsonConvert.DeserializeObject(ex.Response!);
                    Console.WriteLine(Newtonsoft.Json.JsonConvert.SerializeObject(parsed, Newtonsoft.Json.Formatting.Indented));
                } catch {
                    Console.WriteLine(ex.Response);
                }
                Environment.Exit(1);
                return;
            }
            
            var jobId = asyncResponse!.JobId;
            Console.WriteLine($"Job created with ID: {jobId}. Polling for status...");

            while (true)
            {
                try
                {
                    var jobStatus = await client.GetJobStatusAsync(jobId);
                    if (jobStatus.Status == JobStatusResponseStatus.PENDING || jobStatus.Status == JobStatusResponseStatus.RUNNING)
                    {
                        Console.WriteLine($"Status: {jobStatus.Status}");
                    }
                    else
                    {
                        Console.WriteLine("Job completed:");
                        Console.WriteLine(Newtonsoft.Json.JsonConvert.SerializeObject(jobStatus, Newtonsoft.Json.Formatting.Indented));
                        break;
                    }
                    await Task.Delay(1000);
                }
                catch (ApiException ex)
                {
                    Console.WriteLine($"API returned error with status code {ex.StatusCode}:");
                    try {
                        var parsed = Newtonsoft.Json.JsonConvert.DeserializeObject(ex.Response!);
                        Console.WriteLine(Newtonsoft.Json.JsonConvert.SerializeObject(parsed, Newtonsoft.Json.Formatting.Indented));
                    } catch {
                        Console.WriteLine(ex.Response);
                    }
                    Environment.Exit(1);
                    return;
                }
            }
        }
    }
}

File: gridfly-quickstart-java/src/main/java/io/gridfly/quickstart/client/AsyncApiClient.java
package io.gridfly.quickstart.client;

import com.fasterxml.jackson.databind.ObjectMapper;
import io.gridfly.client.generated.ApiClient;
import io.gridfly.client.generated.ApiException;
import io.gridfly.client.generated.ApiResponse;
import io.gridfly.client.generated.api.AsyncJobCreationApi;
import io.gridfly.client.generated.api.AsyncJobStatusApi;
import io.gridfly.client.generated.model.JobCreationResponse;
import io.gridfly.client.generated.model.JobStatusResponse;
import java.io.File;
import java.io.IOException;

public class AsyncApiClient extends BaseApiClient {

  public void generateExcel(String clientId, String clientSecret, File sourceFile)
      throws IOException {

    try {
      ApiClient client = authenticate(clientId, clientSecret);
      AsyncJobCreationApi jobCreationApi = new AsyncJobCreationApi(client);
      AsyncJobStatusApi jobStatusApi = new AsyncJobStatusApi(client);

      ApiResponse<JobCreationResponse> jobCreationResponse =
          jobCreationApi.createAsyncJobWithHttpInfo(true, true, sourceFile);

      boolean finished = false;

      while (!finished) {
        ApiResponse<JobStatusResponse> statusResponse =
            jobStatusApi.getJobStatusWithHttpInfo(jobCreationResponse.getData().getJobId());

        if (statusResponse.getData().getStatus() == JobStatusResponse.StatusEnum.PENDING
            || statusResponse.getData().getStatus() == JobStatusResponse.StatusEnum.RUNNING) {
          System.out.println("Status: " + statusResponse.getData().getStatus());
          try {
            Thread.sleep(1000);
          } catch (InterruptedException e) {
            throw new RuntimeException(e);
          }
        } else {
          ObjectMapper mapper = new ObjectMapper();
          System.out.println(
              "Job completed:\n"
                  + mapper
                      .writerWithDefaultPrettyPrinter()
                      .writeValueAsString(statusResponse.getData()));
          finished = true;
        }
      }
    } catch (ApiException ex) {
      ObjectMapper mapper = new ObjectMapper();
      System.out.println(
          "API returned error with status code "
              + ex.getCode()
              + ":\n"
              + mapper
                  .writerWithDefaultPrettyPrinter()
                  .writeValueAsString(mapper.readValue(ex.getResponseBody(), Object.class)));

      System.exit(1);
    }
  }
}
